virtual_rainforest/core/axes.py

```diff
@@ -140,12 +140,21 @@ def get_validator(axis: str, data: Data, darray: DataArray) -> Optional[Callable
 
     This function iterates over the registered validator functions for a given core axis
     in the AXIS_VALIDATORS registry and matches the data array signatures against the
-    provided signatures. It returns either the appropriate function or None.
+    provided signatures.
+
+    If the input data array uses dimension names that have been registered to validate a
+    particular axis, then a matching validator must be found and an exception is raised
+    if no signature is matched. If the dimension names on the data array do not match
+    any of the registered dimensions, then the function returns None.
 
     Args:
         axis: The core axis to get a validator for
```

Comment:
I think I might be missing something here: what is the difference between an `axis` and a `dimension`? How could an `axis` be related with multiple dimensions?

Reply:
Yes - good point. There is a special case here with the `spatial` axis, which coerces `x` and `y` dimensions onto the `cell_id` spatial axis. In general, there will be only one dimension per axis!

Reply:
Quite confusing, but OK. I'm not sure I have a better solution to this at the moment.